libkbfs/folder_branch_ops.go

```diff
@@ -4669,6 +4683,30 @@ func (fbo *folderBranchOps) onTLFBranchChange(newBID BranchID) {
 	}
 }
 
+func (fbo *folderBranchOps) onMDFlush(bid BranchID, rev MetadataRevision) {
+	ctx, cancelFunc := fbo.newCtxWithFBOID()
+	defer cancelFunc()
+
+	if bid != NullBranchID {
+		fbo.log.CDebugf(ctx, "Ignoring MD flush on branch %v for revision %d",
+			bid, rev)
+		return
+	}
+
+	fbo.log.CDebugf(ctx, "Archiving references for flushed MD revision %d", rev)
+
+	// Get that revision.
+	rmds, err := getMDRange(ctx, fbo.config, fbo.id(), NullBranchID,
```

Comment:
can use the `getSingleMD` helper from your other PR (or port that change into this one, or however you want to do it)